services/web/server/src/simcore_service_webserver/dynamic_scheduler/api.py

```diff
@@ -18,19 +18,30 @@
 from models_library.projects import ProjectID
 from models_library.projects_nodes_io import NodeID
 from models_library.rabbitmq_messages import ProgressRabbitMessageProject, ProgressType
+from models_library.users import UserID
 from pydantic.types import PositiveInt
 from servicelib.progress_bar import ProgressBarData
 from servicelib.rabbitmq import RabbitMQClient, RPCServerError
 from servicelib.rabbitmq.rpc_interfaces.dynamic_scheduler import services
 from servicelib.utils import logged_gather
 
-from ..director_v2.api import list_dynamic_services
 from ..rabbitmq import get_rabbitmq_client, get_rabbitmq_rpc_client
 from .settings import DynamicSchedulerSettings, get_plugin_settings
 
 _logger = logging.getLogger(__name__)
 
 
+async def list_dynamic_services(
+    app: web.Application,
+    *,
+    user_id: UserID | None = None,
```

Comment:
Q: what do these calls return? It is a list of dynamic services in ... 

-1.  `await list_dynamic_services(app)` ??
-2.  `await list_dynamic_services(app, user_id=1)` ??
-3.  `await list_dynamic_services(app, project_id="(some uuid)")` ??
-4.  `await list_dynamic_services(app, user_id=1, project_id="(some uuid)")` ??


Can you also document "known" exceptions if possible?

Reply:
the current running dynamic services. Let's add a small docstring and not keep it as the original.

Reply:
There are no known exceptions

Reply:
What about the question above? what is the expected result on each of the 4 cases above

Reply:
So to answer you questions it's always a list of dynamic services, both legacy and new style filtered by project_id and user_id independently. You can mix them as you please. All above options are accepted.
We generally use options: `1` and `4`. The other two are also supported